test.disabled/ejabberd_tests/tests/push_SUITE.erl

```diff
@@ -0,0 +1,715 @@
+-module(push_SUITE).
+-compile(export_all).
+-include_lib("escalus/include/escalus.hrl").
+-include_lib("common_test/include/ct.hrl").
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("escalus/include/escalus_xmlns.hrl").
+-include_lib("exml/include/exml.hrl").
+
+-define(NS_PUSH,                <<"urn:xmpp:push:0">>).
+-define(NS_XDATA,               <<"jabber:x:data">>).
+-define(NS_PUBSUB_PUB_OPTIONS,  <<"http://jabber.org/protocol/pubsub#publish-options">>).
+-define(PUSH_FORM_TYPE,         <<"urn:xmpp:push:summary">>).
+-define(MUCHOST,                <<"muclight.localhost">>).
+
+-define(MUC_HOST,               <<"muc.localhost">>).
+-define(NS_HTTP_UPLOAD,         <<"urn:xmpp:http:upload">>).
+-define(PUSH_OPTS,
+    [
+        {backend, mnesia}
+    ]).
+
+-import(muc_light_SUITE,
+    [
+        stanza_create_room/3,
+        verify_aff_bcast/2,
+        room_bin_jid/1,
+        gc_message_verify_fun/3
+    ]).
+
+-record(route, {from, to, packet}).
+
+-define(assertReceivedMatch(Expect), ?assertReceivedMatch(Expect, 0)).
+
+-define(assertReceivedMatch(Expect, Timeout),
+    ((fun() ->
+        receive
+            Expect = __Result__ -> __Result__
+        after
+            Timeout ->
+                __Reason__ =
+                receive
+                    __Result__ -> __Result__
+                after
+                    0 -> timeout
+                end,
+
+                __Args__ = [{module, ?MODULE},
+                            {line, ?LINE},
+                            {expected, (??Expect)},
+                            {value, __Reason__}],
+                ct:print("assertReceivedMatch_failed: ~p~n", [__Args__]),
+                erlang:error({assertReceivedMatch_failed, __Args__})
+        end
+      end)())).
+
+
+%%--------------------------------------------------------------------
+%% Suite configuration
+%%--------------------------------------------------------------------
+
+all() ->
+    [
+        {group, disco},
+        {group, toggling},
+        {group, pm_msg_notifications},
+        {group, muclight_msg_notifications}
+    ].
+
+groups() ->
```

Comment:
This set of test cases looks impressive. I have two general comments:
1. This sute is not executed on travis as it is not added to `default.spec`
2. I encourage to run all groups in parallel. `escalus:fresh_story` is your friend here, also you can take a look at `sm_SUITE` or any other suite using `fresh_story`.